lib/ngx/process.md

```diff
@@ -0,0 +1,191 @@
+Name
+====
+
+`ngx.process` - manage the nginx processes for OpenResty/ngx_lua.
+
+Table of Contents
+=================
+
+* [Name](#name)
+* [Status](#status)
+* [Synopsis](#synopsis)
+    * [Enable privileged agent process and get process type](#enable-privileged-agent-process-and-get-process-type)
+* [Methods](#methods)
+    * [type](#type)
+    * [enable_privileged_agent](#enable_privileged_agent)
+* [Community](#community)
+    * [English Mailing List](#english-mailing-list)
+    * [Chinese Mailing List](#chinese-mailing-list)
+* [Bugs and Patches](#bugs-and-patches)
+* [Author](#author)
+* [Copyright and License](#copyright-and-license)
+* [See Also](#see-also)
+
+Status
+======
+
+This Lua module is currently considered experimental.
+The API is still in flux and may change in the future without notice.
+
+Synopsis
+========
+
+Enable privileged agent process and get process type
+-----------------------------------------
+
+```nginx
+# http config
+init_by_lua_block {
+    local process = require "ngx.process"
+
+    -- enable privileged agent process
+    local ok, err = process.enable_privileged_agent(true)
+    if not ok then
+        ngx.log(ngx.ERR, "enable privileged agent failed error:", err)
+    end
+
+    -- output process type
+    ngx.log(ngx.INFO, "process type: ", process.type(true))
+}
+
+init_worker_by_lua_block {
+    local process = require "ngx.process"
+    ngx.log(ngx.INFO, "process type: ", process.type(true))
+}
+
+server {
+    # ...
+    location = /t {
+        content_by_lua_block {
+            local process = require "ngx.process"
+            ngx.say("process type: ", process.type(true))
+        }
+    }
+}
+
+```
+
+The example config above produces an output to `error.log` when
+server starts:
+
+```
+[lua] init_by_lua:11: process type: master process
+[lua] init_worker_by_lua:3: process type: privileged agent process
+[lua] init_worker_by_lua:3: process type: worker process
```

Comment:
I'm afraid the output is stale. We no longer have the " process" suffix in the `process.type` return values.